spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -363,6 +363,58 @@ def get_unit_max_channels(recording, sorting, unit_ids=None, peak='both', mode='
     else:
         return max_list
 
+def get_unit_amplitudes(recording, sorting, unit_ids=None, grouping_property=None, channels=None,
+                        ms_before=3., ms_after=3., dtype=None, max_num_waveforms=np.inf,
+                        save_as_features=True, compute_property_from_recording=False, verbose=False):
+    '''
+    Computes the spike amplitudes from a recording and sorting extractor.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor
+    sorting: SortingExtractor
+        The sorting extractor
+    unit_ids: list
+        List of unit ids to extract waveforms
+    grouping_property: str
+        Property to group channels. E.g. if the recording extractor has the 'group' property and 'grouping_property' is
+        'group', then waveforms are computed group-wise.
+    channels: list
+        List of channels to compute waveforms from
+    ms_before: float
+        Time period in ms to cut waveforms before the spike events
+    ms_after: float
+        Time period in ms to cut waveforms after the spike events
+    dtype: dtype
+        The numpy dtype of the waveforms
+    max_num_waveforms: int
+        The maximum number of waveforms to extract (default is np.inf)
+    save_as_features: bool
+        If True (default), waveforms are saved as features of the sorting extractor object
+    compute_property_from_recording: bool
+        If True and 'grouping_property' is given, the property of each unit is assigned as the corresponding propery of
+        the recording extractor channel on which the average waveform is the largest
+    verbose: bool
+        If True output is verbose  
+
+    Returns
+    -------
+    amplitudes: np.array
+        List of np.array (n_spikes, 1) containing extracted amplitudes for each spike
+    '''  
+
+    spike_times = np.array([])
+
+    for i_u, id in enumerate(sorting.get_unit_ids()):
+        st = sorting.get_unit_spike_train(id)
+        spike_times = np.concatenate((spike_times, np.array(st)))
+
+    #Dummy amps for now
+    amplitudes = np.ones((len(spike_times), 1), dtype='int16')
```

Comment:
this can be computed for each spike. You can get waveforms and templates and compute the ratio between each waveform (or a subset of those) and the template. That is similar to the KS amplitudes (it could be don in the phy output as well :) )

Reply:
That can be done eventually. First priority is to get every metric integrated.

Reply:
:+1: